src/models/user_staking.ts

```diff
@@ -0,0 +1,112 @@
+import * as R from 'ramda';
+
+type DefaultDelegationType = {
+  validatorAddress: string;
+  amount: {
+      denom: string;
+      amount: number;
+  }
+  commission: number;
+  votingPower: number;
+}
+
+type DefaultUnbondingType = {
+  validatorAddress: string;
+  amount: {
+      denom: string;
+      amount: number;
+  }
+  height?: number;
+  expectedTime: string;
+}
+
+type DefaultRedelegationType = {
+  srcValidatorAddress: string;
+  dstValidatorAddress: string;
+  delegatorAddress: string;
+  amount: {
+      denom: string;
+      amount: number;
+  }
+  height: number;
+  expectedTime: string;
+}
+
+class UserStaking {
+  public delegations: DefaultDelegationType[];
+  public rewards?: {
+    validatorAddress: string;
+    delegatorAddress: string;
+    amount: {
+      denom: string;
+      amount: number;
+    }}[];
+  public redelegations: DefaultRedelegationType[];
+  public unbonding: DefaultUnbondingType[];
+  public totalVotingPower: number;
```

Comment:
`DefaultDelegationType` `DefaultRedelegationType` `DefaultUnbondingType` are only used once 

two options:

1)  declare a type for `rewards` too for consistent code
2) remove all the `types` and directly declare it like `rewards`